command.go

```diff
@@ -479,41 +491,67 @@ func (c *Command) exec(ctx context.Context, opts *CreateOptions, idx int) error
 
 	addOutOp := func(msg message.Fields) message.Fields { return msg }
 	var err error
-	var newProc Process
 	// TODO: the logic below this is not strictly correct if, for example,
 	// Output is redirected to Error and Error has been defined.
 	if opts.Output.Output == nil {
 		var out bytes.Buffer
 		opts.Output.Output = &out
 		opts.Output.Error = &out
-		newProc, err = c.makep(ctx, opts)
+		proc, err := c.makep(ctx, opts)
 		if err != nil {
 			return errors.Wrapf(err, "problem starting command")
 		}
+		c.procs = append(c.procs, proc)
 
-		c.procIDs = append(c.procIDs, newProc.ID())
 		addOutOp = func(msg message.Fields) message.Fields {
 			getLogOutput(out.Bytes())
 			return msg
 		}
 	} else {
-		newProc, err = c.makep(ctx, opts)
+		proc, err := c.makep(ctx, opts)
 		if err != nil {
 			return errors.Wrapf(err, "problem starting command")
 		}
-
-		c.procIDs = append(c.procIDs, newProc.ID())
+		c.procs = append(c.procs, proc)
 	}
 
 	if !c.runBackground {
-		_, err = newProc.Wait(ctx)
-		msg["err"] = err
+		waitCatcher := grip.NewBasicCatcher()
+		for _, proc := range c.procs {
+			_, err = proc.Wait(ctx)
+			waitCatcher.Add(errors.Wrapf(err, "error waiting on process '%s'", proc.ID()))
+		}
+		msg["err"] = waitCatcher.Resolve()
 	}
-
 	grip.Log(c.priority, addOutOp(msg))
+
 	return errors.WithStack(err)
 }
 
+// Wait returns the exit code and error waiting for the underlying process to
+// complete.
+// For processes running in parallel, it only returns a zero exit code
```

Comment:
I think this is wrong because there's no parallelism here?

Reply:
This doc comment should still be true. I meant that Commands run with `RunParallel()` instead of `Run()` will behave this way. I clarified the doc comment to say this.